### ISSUE TITLE
Loading bar http client not working with async interceptor

### ISSUE THREAD

**Issue by FLCreative** [aitboudad/ngx-loading-bar#65]
Hello,

I have updated my interceptor to get my token async, but now the loading bar is not displayed on request.

My project works on Angular 5 latest version.

This is my interceptor, can you help me ?

```
  intercept(req: HttpRequest<any>, next: HttpHandler) {
    console.log('interceptor called');
    if (!req.headers.has('Content-Type')) {
      req = req.clone({ headers: req.headers.set('Content-Type', 'application/json') });
    }

    req = req.clone({ setHeaders: {Accept: 'application/json', 'X-Requested-With': 'XMLHttpRequest'} });

    return Observable.fromPromise(this.addToken(req, next));
  }

  private async addToken(req: HttpRequest<any>, next: HttpHandler): Promise<HttpEvent<any>>  {
    const user: User = await this.fAuth.auth.currentUser;

    if (user != null) {
      await user.getIdToken().then(token => {
        req = req.clone({ setHeaders: {Authorization: token} });
      });

    }

    return next.handle(req).toPromise();
  }
```

**Comment by aitboudad** [comment]
fixed in `1.3.1`

**Comment by FLCreative** [comment]
Thanks for the quick fix, working great ;)
